src/error.rs

```diff
@@ -128,3 +128,22 @@ impl StdError for WrongGeneration {
          (e.g. because the entity has been deleted)"
     }
 }
+
+
+/// An error type which cannot be instantiated.
+/// Used as a placeholder for associated error types if
+/// something cannot fail.
+#[derive(Debug)]
+pub enum NoError {}
```

Comment:
Should be `PhantomError`

Reply:
Should be `!` but it isn't stable yet.
Why `PhantomError` anyway?

Reply:
Well, first, because you are implementing `Error` for `NoError`, which already looks like a hack. Secondly, because the "phantom" is rather established for such things that can't be instantiated, so `PhantomError` is crystal clear here.

Reply:
`PhantomData` is unit type and can be instantiated. And it is an example from `std`.
IDK any other examples.

Reply:
I don't have a strong opinion here, @kvark do you insist on this request?

Reply:
Nope. I think I might have confused a bit the semantics of phantom types. Pretty sure I saw them used in the context where one couldn't create instances (`enum MyPhantomType {}`), but I don't want to push hard on that here. It's fine to ship as is.